### ISSUE TITLE
Trying to render the same plot twice is failing

### ISSUE THREAD

**Issue by damianavila** [bokeh/bokeh#2415]
Test case:

Not working (same figure): 
```python
import numpy as np

from bokeh.plotting import figure, show

x = np.linspace(0, 4*np.pi, 200)
y = np.sin(x)

p = figure(title="simple line example")
p.line(x,y, color="#2222aa", line_width=2)

p.ygrid[0].ticker.desired_num_ticks = 20

import jinja2
from bokeh.embed import components

template = jinja2.Template("""
<!DOCTYPE html>
<html lang="en-US">

<link
    href="http://cdn.pydata.org/bokeh/release/bokeh-0.12.2.min.css"
    rel="stylesheet" type="text/css"
>
<script 
    src="http://cdn.pydata.org/bokeh/release/bokeh-0.12.2.min.js"
></script>

<body>

    <h1>Hello Bokeh!</h1>
    
    <p> Below is a simple plot of stock closing prices </p>
    
    {{ script }}
    
    {{ div }}

    <p> Below is a simple plot of stock closing prices </p>

    {{ script2 }}
    
    {{ div2 }}

</body>

</html>
""")

script, div = components(p)
script2, div2 = components(p)

html = template.render(script=script, div=div, script2=script2, div2=div2)

with open("test.html", "w+") as f:
    f.write(html)
```

Working (different figures):

```python
import numpy as np

from bokeh.plotting import figure, show

x = np.linspace(0, 4*np.pi, 200)
y = np.sin(x)

p = figure(title="simple line example")
p.line(x,y, color="#2222aa", line_width=2)

p.ygrid[0].ticker.desired_num_ticks = 20

q = figure(title="second")
q.line(x,y, color="black", line_width=5)

import jinja2
from bokeh.embed import components

template = jinja2.Template("""
<!DOCTYPE html>
<html lang="en-US">

<link
    href="http://cdn.pydata.org/bokeh/release/bokeh-0.9.0.min.css"
    rel="stylesheet" type="text/css"
>
<script 
    src="http://cdn.pydata.org/bokeh/release/bokeh-0.9.0.min.js"
></script>

<body>

    <h1>Hello Bokeh!</h1>
    
    <p> Below is a simple plot of stock closing prices </p>
    
    {{ script }}
    
    {{ div }}

    <p> Below is a simple plot of stock closing prices </p>

    {{ script2 }}
    
    {{ div2 }}

</body>

</html>
""")

script, div = components(p)
script2, div2 = components(q)

html = template.render(script=script, div=div, script2=script2, div2=div2)

with open("test.html", "w+") as f:
    f.write(html)
```

**Comment by damianavila** [comment]
AFAIK, @bryevdv is working in a fix...

**Comment by damianavila** [comment]
hey @bryevdv, IIRC, this was solved, but I don't remember where? Do you have any record of this, so I can link the PR? thanks!

**Comment by bryevdv** [comment]
I don't recall doing anything specifically for this issue.

**Comment by damianavila** [comment]
Looking int the past, you solve the issue in the bokeh notebooks (tutorial) making a make_figure function... but I guess this is just a workaround: http://nbviewer.ipython.org/github/bokeh/bokeh-notebooks/blob/master/tutorial/05%20-%20sharing.ipynb

**Comment by bryevdv** [comment]
Oh, yes absolutely, not a real fix.

**Comment by damianavila** [comment]
Previous milestone: 0.9.1

**Comment by bryevdv** [comment]
this was fixed at some pointed (tested working with `0.12.2`)
